byway/src/main/java/com/google/sps/data/Trip.java

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.Collection;
+import java.util.ArrayList;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A class to make a Trip type, containing
+ * a specific id, destinations, interests and routes 
+ * to be used when updating a trip.
+ */
+public final class Trip {
+
+  private final String id;
+  private final String start;
+  private final ArrayList<String> destinations;
+  private final ArrayList<String> interests;
+  private final ArrayList<String> route;
+
+  public Trip(String id, String start, Collection<String> destinations,
```

Comment:
It would also be nice to have a constructor that takes in an entity object returned from datastore and constructs a Trip. Or you could have a separate static helper function that builds a Trip from a datastore return object. Something like 
```java
createTripFromEntity(Entity tripEntity){
  id = TripEntity.getProperty("id");
  ...
  return new Trip(id, start, destinations, ....);
}
```

Reply:
Noted. I created the helper functions within each class along with checks to make sure that
a) the entity passed is is not null itself, and
b) each property required to make a User/Trip class was valid/not null

Fixed

Reply:
Nice

Reply:
Nice!